File: flutter_app/lib/models/company_profile.dart
class CompanyProfile {
  final int? id;
  final String companyName;
  final String? address;
  final String? contactEmail;
  final String? contactPhone;
  final String? taxId;

  CompanyProfile({
    this.id,
    required this.companyName,
    this.address,
    this.contactEmail,
    this.contactPhone,
    this.taxId,
  });

  factory CompanyProfile.fromJson(Map<String, dynamic> json) {
    return CompanyProfile(
      id: json['id'],
      companyName: json['companyName'] ?? '',
      address: json['address'],
      contactEmail: json['contactEmail'],
      contactPhone: json['contactPhone'],
      taxId: json['taxId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'companyName': companyName,
      'address': address,
      'contactEmail': contactEmail,
      'contactPhone': contactPhone,
      'taxId': taxId,
    };
  }
}

File: backend/src/main/java/com/isravel/workhub/settings/CompanyProfile.java
package com.isravel.workhub.settings;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "company_profiles")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CompanyProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "company_name", nullable = false)
    private String companyName;

    @Column(name = "address")
    private String address;

    @Column(name = "contact_email")
    private String contactEmail;

    @Column(name = "contact_phone")
    private String contactPhone;

    @Column(name = "tax_id")
    private String taxId;

    @Column(name = "day_boundary")
    private java.time.LocalTime dayBoundary = java.time.LocalTime.of(6, 0);
}

